File: discover_javascript/write_them_to_disc.js
var https = require('https');

var options = {
    hostname: 'api.github.com',
    path: '/search/repositories?q=language:javascript&sort=stars&order=desc',
    headers: {
        'User-Agent': 'Holberton_School',
        'Authorization': 'token ' + process.env.TOKEN
    }
};

var req = https.get(options, function (res) {
    const chunks = [];
    var jsonString = '';
    res.on('data', (chunk) => {
        chunks.push(chunk);
    });
    res.on('end', () => {
        jsonString = chunks.join('');
        var filesystem = require('fs');
        filesystem.writeFile("/tmp/25", jsonString, function(err){
            if(err){
                return console.log(err);
            }
            else{
                return console.log('The file was saved!');
            }
        });
    });
});

req.on('error', function (e) {
    console.error(e);
});

File: discover_ruby/write_them_to_disc.rb
require 'httpclient'

extheaders = {
  'User-Agent' => 'Holberton School',
  'Authorization' => 'token f11ca8ea6bf7bf5456f862debf389d94961bdb15'
}

request = HTTPClient.new
File.open('/tmp/25','w') do |file|
	file.write(request.get_content('https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc', query = nil, extheaders))
end
puts "The file was saved!"
